invoice/src/main/java/org/killbill/billing/invoice/tree/AccountItemTree.java

```diff
@@ -160,7 +161,13 @@ public void mergeWithProposedItems(final List<InvoiceItem> proposedItems) {
         for (SubscriptionItemTree tree : subscriptionItemTree.values()) {
             final List<InvoiceItem> simplifiedView = tree.getView();
             if (simplifiedView.size() > 0) {
-                result.addAll(simplifiedView);
+                for (final InvoiceItem i : simplifiedView) {
+                    // Filter out $0 REPAIR_ADJ - these items make a lot of sense conceptually for us but may confuse users...
```

Comment:
I went back and forth but I'm now thinking we should keep them after all to be consistent with https://github.com/killbill/killbill/issues/667.

It could be confusing but it's also such a corner case scenario (doing full item adjustment and a repair).